app/services/address_geocoder.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class AddressGeocoder
+  def initialize(address)
+    @address = address
+  end
+
+  def geocode
+    latitude, longitude = Geocoder.coordinates(geocode_address)
+
+    return unless latitude.present? && longitude.present?
+
+    address.update(latitude: latitude, longitude: longitude)
+  end
+
+  private
+
+  attr_reader :address
+
+  def geocode_address
+    [
+      address.address1,
+      address.address2,
+      address.zipcode,
+      address.city,
+      address.country.andand.name,
+      address.state.andand.name
+    ].select(&:present?).join(', ')
+  end
```

Comment:
Rubocop complains about the complexity here. I find it a bit strict but we could split this into two methods:
```suggestion
  def geocode_address
    address_parts.select(&:present?).join(', ')
  end

  def address_parts
    [
      address.address1,
      address.address2,
      address.zipcode,
      address.city,
      address.country&.name,
      address.state&.name
    ]
  end
```

Reply:
This seems a bit strict to me too so I left that one out if that's ok